[merge-mining] Add Tari aux chain data to getlastblockheader

sdbondi opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Merge mining proxy now includes tari block header data as aux chain data
    on getlastblockheader call.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Pool uses this to know if a new block template should be requested.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Tested on nodejs-pool fork

sdbondi committed:
[merge-mining] Submit_block AUX data includes tari block hash

Calls to submit_block return the tari hash if the block is accepted by
the Tari blockchain.

- [fix] Return the correct response when tari's get_header_by_hash GRPC call
  returns "not found"
- [feat] get_block_template includes tari block height in aux chain data
- [fix] get_block_header_by_hash should not forward the request to tari
  base node if block was found by monerod.
- [fix] Add correct `Host` http header on proxied requests. Some public monero
  daemons require this.
- [chore] Minor request logging improvements

StriderDM commented:
Please rebase against https://github.com/tari-project/tari/pull/2699 after it merges.

sdbondi committed:
[merge-mining] Add Tari aux chain data to getlastblockheader

Merge mining proxy now includes tari block header data as aux chain
data on getlastblockheader call.
